### DISCUSSION TITLE
how to get faces from edges

### DISCUSSION THREAD

**Discussion by weiminw** [pyg-team/pytorch_geometric#5432] [Q&A]
I got data from ShapeNet dataset 
``` 
dataset = ShapeNet(root="/Users/wangweimin/Workspaces/pytorch-geometric-datasets/SHAPENET", pre_transform=T.KNNGraph(k=2))
data = dataset[0]
```
but is there any way to get faces from data.edge_index

**Comment by rusty1s** [comment]
See https://github.com/pyg-team/pytorch_geometric/discussions/5430

**Reply by weiminw** [reply]
> See #5430

Great thank to your help.well, Is there any function provided in PyG can transform data.edge_index matrix to faces matrix ?

**Reply by rusty1s** [reply]
No, such a function does not exist since generally not every graph can be converted into a face matrix.
